spot-client/src/spot-tv/native-functions/native-controller/native-controller.js

```diff
@@ -0,0 +1,45 @@
+import { logger } from 'common/logger';
+
+/**
+ * Implements a controller that utilizes the event emitter of the native (Electron) process if available.
+ *
+ * Using it will enable SpotTVs running wrapped in Electron access OS-native functions.
+ */
+class NativeControntroller {
+    /**
+     * Some.
+     */
+    constructor() {
+        // eslint-disable-next-line
+        this.ipcRenderer = this.isAvailable() && __non_webpack_require__('electron').ipcRenderer;
+    }
+
+    /**
+     * Returns true if the native functions are available (electron env.), false otherwise (web env.).
+     *
+     * @returns {boolean}
+     */
+    isAvailable() {
+        return window && window.process && window.process.emit;
+    }
+
+    /**
+     * Sends a message to the native process.
+     *
+     * @param {string} command - The command identifyer.
+     * @param {Object} args - Args of the command, if any.
+     * @returns {void}
+     */
+    sendMessage(command, args) {
+        if (this.isAvailable()) {
+            this.ipcRenderer.send('native-command', {
+                command,
+                args
+            });
```

Comment:
Is there a way to get confirmation back that the command finished or succeeded? I don't think it's needed now (and to be fair may never be needed) but it'd be nice to know we can use something like a promise if needed.

Reply:
Yes, we can do that. Although I'd be worried about promises that may never be resolved, for example when we try to send a message to a non-electron client. Although, since it's not really an async flow, I'd go with a message-reply approach with listener for now, if need be

Reply:
This is something we should do when there is a need for that, I think